mantidproject/mantid
# Improve spectra selection within Sliceviewer downsampling

**Description of work.**

This PR improves the spectra selection within the sliceviewer when downsampling the image. This avoids missing high counting detectors when the sliceviewer is first opened. To solve this issue, a maxpooling approach is used, which ensures that within each sampling range (ymin, ymax) the highest counting detector is selected.



**To test:**

1. Code review
2. As per the attatched issue, do the following:
3. Load MAR27283.nxs 
4. Open sliceviewer on the workspace (**note**: on Debug builds this appears to take a while...)
5. The high counting spectra at 794 should now be shown (previously it wasn't)



Fixes #29237 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
